fourth.py

```diff
@@ -6,15 +6,29 @@ def fibo_fun(n):
 
 print(fibo_fun(8))
 
+# to reduece time complexity
 
-Numbers = [23, 44, 5, 67, 1, 1, 2, 4, 5]
+cache_store = {0:1,1:1}
+
+def fibo_fun_witch_cache(n):
+    if n in {0,1}:
+        return cache_store[n]
+    else:
+        cache_store[n] = fibo_fun_witch_cache(n-1)+fibo_fun_witch_cache(n-2)
+        return cache_store[n]
+
+fibo_nums = [fibo_fun_witch_cache(n) for n in range(8)]
```

Comment:
Get input from user and whatever input user provides, return the fibonacci value for it.

For example, 
user enters n = 10
Output: value of fibonacci at 10th step

pass n = 90
I am sure your code will not give the answer. That's why I told you to used memoization. Try again for it, if you didn't find any clue, then leave it for now.